src/Service.Tests/GraphQLBuilder/NestedMutationBuilderTests.cs

```diff
@@ -0,0 +1,407 @@
+        /// leak. These linking object definitions are only used to generate the final source->target linking object definitions for entities
+        /// having an M:N relationship between them.
+        /// </summary>
+        /// <param name="sourceEntityName">Name of the source entity for which the configuration is provided in the config.</param>
+        /// <param name="targetEntityName">Name of the target entity which is related to the source entity via a relationship defined in the 'relationships'
+        /// section in the configuration of the source entity.</param>
+        [DataTestMethod]
+        [DataRow("Book", "Author", DisplayName = "Validate absence of linking object for Book->Author M:N relationship")]
+        [DataRow("Author", "Book", DisplayName = "Validate absence of linking object for Author->Book M:N relationship")]
+        [DataRow("BookNF", "AuthorNF", DisplayName = "Validate absence of linking object for BookNF->AuthorNF M:N relationship")]
+        [DataRow("AuthorNF", "BookNF", DisplayName = "Validate absence of linking object for AuthorNF->BookNF M:N relationship")]
+        public void ValidateAbsenceOfLinkingObjectDefinitionsInObjectsNodeForMNRelationships(string sourceEntityName, string targetEntityName)
+        {
+            string linkingEntityName = GraphQLUtils.GenerateLinkingEntityName(sourceEntityName, targetEntityName);
+            ObjectTypeDefinitionNode linkingObjectTypeDefinitionNode = GetObjectTypeDefinitionNode(linkingEntityName);
+
+            // Assert that object definition for linking entity/table is null here.
+            // The object definition being null here implies that the object definition is not exposed in the objects node.
+            Assert.IsNull(linkingObjectTypeDefinitionNode);
+        }
+
+        /// <summary>
+        /// Test to validate that that we create a source -> target linking object definition for every pair of (source, target) entities which
+        /// are related via an M:N relationship.
+        /// </summary>
+        /// <param name="sourceEntityName">Name of the source entity for which the configuration is provided in the config.</param>
+        /// <param name="targetEntityName">Name of the target entity which is related to the source entity via a relationship defined in the 'relationships'
+        /// section in the configuration of the source entity.</param>
+        [DataTestMethod]
+        [DataRow("Book", "Author", DisplayName = "Validate presence of source->target linking object for Book->Author M:N relationship")]
+        [DataRow("Author", "Book", DisplayName = "Validate presence of source->target linking object for Author->Book M:N relationship")]
+        [DataRow("BookNF", "AuthorNF", DisplayName = "Validate presence of source->target linking object for BookNF->AuthorNF M:N relationship")]
+        [DataRow("AuthorNF", "BookNF", DisplayName = "Validate presence of source->target linking object for AuthorNF->BookNF M:N relationship")]
```

Comment:
are each of these test cases validating different types of "M:N" relationships? I can't tell what the difference is between each data row besides entity name.
e.g. for data row 1, what about the entities book / author make them in scope of the validation? config relationship definitions? relationships resolved from db metadata? Linked table?

Reply:
Yeah, there is absolutely no difference in terms of what is required from the perspective of test coverage. I'll keep only one.